server/src/main/java/io/crate/replication/logical/action/PublicationsStateAction.java

```diff
@@ -183,37 +223,70 @@ protected ClusterBlockException checkBlock(Request request,
                         }
                         return false;
                     })
+                    .filter(t -> userCanPublish(t.ident(), publicationOwner))
+                    .filter(t -> subscriberCanRead(t.ident(), subscriber))
                     .map(RelationInfo::ident)
                     .toList();
             } else {
-                relationNames = publication.tables();
+                // No need to call userCanPublish() since for the pre-defined tables case this check was already done on the publication creation.
+                relationNames = publication.tables()
+                    .stream()
+                    .filter(t -> subscriberCanRead(t, subscriber))
+                    .toList();
             }
             return relationNames;
         }
+
+        private static boolean subscriberCanRead(RelationName relationName, User subscriber) {
+            return subscriber.hasPrivilege(Privilege.Type.DQL, Privilege.Clazz.TABLE, relationName.fqn(), Schemas.DOC_SCHEMA_NAME);
+        }
+
+        private static boolean userCanPublish(RelationName relationName, User publicationOwner) {
+            for (Privilege.Type type: READ_WRITE_DEFINE) {
+                // This check is triggered only on ALL TABLES case.
+                // Required privileges correspond to those we check for the pre-defined tables case in AccessControlImpl.visitCreatePublication.
+
+                // Schemas.DOC_SCHEMA_NAME is a dummy parameter since we are passing fqn as ident.
+                if (!publicationOwner.hasPrivilege(type, Privilege.Clazz.TABLE, relationName.fqn(), Schemas.DOC_SCHEMA_NAME)) {
+                    return false;
+                }
+            }
+            return true;
+        }
     }
 
     public static class Request extends MasterNodeReadRequest<Request> {
 
         private final List<String> publications;
+        private final String subscriber;
```

Comment:
Would change it to avoid confusion about subscriber == subscribingCluster/node.

```suggestion
        private final String subscribingUserName;
```